doc/source/examples/multiphysics/dam-break-VOF/dam-break-VOF.rst

```diff
@@ -0,0 +1,255 @@
+        end
+    end
+
+Two fluids are present in this simulation, hence in the ``physical 
+properties`` subsection, their physical properties should be defined:
+
+
+.. code-block:: text
+
+    #---------------------------------------------------
+    # Physical Properties
+    #---------------------------------------------------
+    subsection physical properties
+        set number of fluids     = 2
+        subsection fluid 0
+            set density              = 0.02
+            set kinematic viscosity  = 0.1
+        end
+        subsection fluid 1
+            set density              = 0.9982
+            set kinematic viscosity  = 0.01
+        end
+    end
+
+In the ``mesh adaptation subsection``, adaptive mesh refinement is 
+defined for ``velocity``. ``min refinement level`` and ``max refinement 
+level`` are 4 and 5, respectively.
+
+.. code-block:: text
+
+    #---------------------------------------------------
+    # Mesh Adaptation
+    #---------------------------------------------------
+    subsection mesh adaptation
+        set type                    = kelly
+        set variable                = velocity
+        set fraction type           = fraction
+        set max refinement level    = 5
+        set min refinement level    = 4
+        set frequency               = 5
+        set fraction refinement     = 0.2
+        set fraction coarsening     = 0.01
+    end
+
+
+
+*Call the gls_navier_stokes_2d by invoking:*  
+
+``mpirun -np 2 ./{path-to-lethe-build-dir}/applications/gls_navier_stokes_2d/gls_navier_stokes_2d gls_VOF_dam-break_Martin_and_Moyce.prm``
+
+*to run the simulation using two CPU cores.* (Feel free to use more)
+
+
+.. warning:: 
+    The code will compute :math:`100,000+` dofs for :math:`620+` time 
+    iterations. Make sure to compile lethe in `Release` mode and 
+    run in parallel using mpirun 
+
+
+
+
+-------
+Results
+-------
+The following image shows the screenshots 
+of the simulation at :math:`0`, :math:`1.1`, :math:`3`, and :math:`4` seconds
+
+
+
+
+.. image:: images/time-shots.png
+    :alt: time-shots
+    :align: center
+
+A python post-processing code `(gls_VOF_dam-break_Martin_and_Moyce.py)` 
+is added to the example folder for post-processing the results.
+Run `python3 ./Dambreak_2d_lethe.py ./Output` to execute this 
+post-processing code, where `./Output` is the directory that 
+contains the simulation results. In post-processing, the maximum 
+dimensionless lateral position of the liquid phase is tracked 
+through time and compared with the experiments of Martin and Moyce
+(1952). The following figure shows the result of
+the post-processing:
+
+.. image:: images/xmax_t.png
+    :alt: xmax_t
+    :align: center
+
+
+As mentioned previously, this simulation uses adaptive mesh
+refinement. The following image shows the mesh and the position of
+the interface at :math:`4` seconds. The mesh refinement detects 
+and refines the meshes on the interface.
```

Comment:
Clearly the coarsening is not agressive enough, you see cells remaining :).

Reply:
and with refinement with the phase parameter it will be even more beautiful :)

Reply:
Refinement with the phase fraction does not work in this example. It does not refine the moving side of the interface. :)

Reply:
Have you tried with   
`set fraction type		= fraction`
`set fraction refinement = 0.8`

? (see the new doc https://lethe-cfd.github.io/lethe/parameters/cfd/mesh_adaptation_control.html ;) )

Reply:
if you put `set fraction refinement = 0.1` as you would for a velocity mesh adaptation, it is normal that it does not want to refine the most "blurry" parts of the interface, which are the moving parts

Reply:
yes, it still does not work with fraction refinement = 0.8 :)
![image](https://user-images.githubusercontent.com/29518349/150181216-b4f90552-1026-4a18-baaa-f0100ccd0099.png)

Reply:
In this case you can try 0.9 or 0.95 for example, anyway the rest of the cells will be far below this margin of error.

Reply:
I agree that it needs time to catch up with fast moving interface, but it still is working properly. I think you base mesh is too thin for you to really see it, so here is what I have with these parameters (much coarser mesh at the beginning, so really quicker simulation):

`#---------------------------------------------------
# Mesh
#---------------------------------------------------
subsection mesh
        set type = dealii
        set grid type = subdivided_hyper_rectangle
        set grid arguments = 14, 10 : 0, 0 : 14, 10 : true
        set initial refinement = 3
end

#---------------------------------------------------
# Mesh Adaptation
#---------------------------------------------------
subsection mesh adaptation
  set type                    = kelly
  set variable                = phase
  set fraction type           = fraction
  set max refinement level    = 5
  set min refinement level    = 2
  set frequency               = 1
  set fraction refinement     = 0.9
  set fraction coarsening     = 0.1
end`

At iteration 0:
![image](https://user-images.githubusercontent.com/72091781/150186645-1a60c8b9-d10a-4da6-9a5a-165698b05459.png)

At iteration 50 (output 10):
![image](https://user-images.githubusercontent.com/72091781/150187990-60ed10d9-e67b-4369-9437-2246bf9421b8.png)

As for performance, refining at frequency = 1 have the simulation plateau at about
`   Number of active cells:       5159
   Number of degrees of freedom: 16707
   Volume of triangulation:      140
   Number of VOF degrees of freedom: 5569
`

So it is notably quicker (on my laptop, with `mpirun -np 5`, each transient iteration takes about 5-10 sec). 

I could not launch your prm on my machine (10th gen i7, 16Go RAM), so I am concerned that users might find it impossible to launch this example as is. It will surely decrease a bit the agreement with the article, for this I think your prm is best as the mesh is really thin, but I think it would be better for documentation purposes to have an example launchable on a normal machine.

I will leave the final decision to the consensus!

Reply:
The original prm file needs approximately 1 hour to finish on 8 cores. As this is not an application test, but an example, its goal is not to be fastly finished on laptops. We use examples to show how to set up simulations and show the accuracy of the solvers compared with experiments or other software results. I launched a simulation with the changes you made on the mesh and refinements. It's significantly faster, but still, with a fraction refinement = 0.9, the right edge of the interface is not covered with refined cells at several time-steps:
![image](https://user-images.githubusercontent.com/29518349/150191152-65ebde6f-0ae0-43ef-956e-1f56d60b0ac1.png)

I think this may cause problems when we use the post-processing code to obtain the maximum value of the interface since the post-processing code calculates the maximum value at `y` = 0. Hence, the blurry edge of the interface should lead to underestimation of the experimental values.

I agree to leave the final decision to the consensus.

Reply:
It is not "fastly finished on laptops" with the changes I propose, it is still takes about 1 hour on my laptop with `mpirun -np 5`. I really agree that the agreement with the article will not be as good, obviously, but I would be concerned that the initial prm is not even launchable for a normal user (it was even stuck at initialization on my laptop... which is again pretty decent for computational work). 

Maybe the two could co-exist? Like, we provide a .prm that can be launched more easily by interested users, so that they can play around with the example, but we also give your prm and results after, to demonstrate the code abilities with a fine mesh and a good agreement with literature. This way we can also add a section to explain that for fast moving free surface, adaptation on the velocity catches well the interface at the front, but that for catching the interface globally it is faster to have adaptation on the phase. What do you think?

(sorry Abbas for the mess ^^" )

Reply:
Very good idea.

Reply:
@AbbasBalloutRe do not hesitate if you want me to help you rewriting the example!

Reply:
This is the comparison with literature I get from the "light simulation" prm file :
![image](https://user-images.githubusercontent.com/72091781/150208148-7000dd28-5ba3-493b-8aa1-6910d49d2674.png)

It will be worth putting it also to explain the difference with the "accurate simulation" you put in place Shahab. This was already an impressive example, all of this together will make it super hot :)

Reply:
I just saw this one. Sorry